src/daft-sql/src/planner.rs

```diff
@@ -79,11 +79,11 @@ struct OrderByExprs {
 /// TODO consolidate SQLPlanner state to the single context.
 /// TODO move bound_ctes into per-planner scope since these are a scoped concept.
 #[derive(Default)]
-struct PlannerContext {
+pub(crate) struct PlannerContext {
     /// Session provides access to metadata and the path for name resolution.
     /// TODO move into SQLPlanner once state is flipped.
     /// TODO consider decoupling session from planner via a resolver trait.
-    session: Rc<Session>,
+    pub(crate) session: Rc<Session>,
```

Comment:
Why more public? I feel making things public (even if just `pub(crate)`) to get things done is usually a hack or indicates poor design somewhere. Rather than opening up these types perhaps we can make them better as their usage expands? This should give us more flexibility to modify implementations while SQL is still nascent